src/main/java/seedu/contax/logic/parser/BatchCommandParser.java

```diff
@@ -19,19 +21,32 @@
     public BatchCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_SEARCH_TYPE, PREFIX_EQUALS);
-
+                ArgumentTokenizer.tokenize(args,
+                        PREFIX_SEARCH_TYPE, PREFIX_EQUALS, PREFIX_START_WITH, PREFIX_END_WITH);
         String commandInput = argMultimap.getPreamble();
-        if (!argMultimap.arePrefixesPresent(PREFIX_SEARCH_TYPE, PREFIX_EQUALS)
-            || argMultimap.getPreamble().isEmpty()) {
+        String equalsText = "";
+        String startText = "";
+        String endText = "";
+        if (!argMultimap.arePrefixesPresent(PREFIX_SEARCH_TYPE) || argMultimap.getPreamble().isEmpty()
+            || (!argMultimap.arePrefixesPresent(PREFIX_EQUALS)
+                && !argMultimap.arePrefixesPresent(PREFIX_START_WITH)
+                && !argMultimap.arePrefixesPresent(PREFIX_END_WITH))) {
```

Comment:
The method `arePrefixesPresent` takes multiple arguments, it does not need to be this cluttered.

Reply:
thx will change

Reply:
Alright, nevermind then, then a `getValue().isPresent()` check would have sufficed.

Reply:
yes but anyway the effect shd be same